sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubBufferedPartitionProducer.java

```diff
@@ -15,87 +16,129 @@
 import reactor.core.publisher.BaseSubscriber;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoSink;
 import reactor.core.publisher.Sinks;
 import reactor.core.scheduler.Schedulers;
 import reactor.util.concurrent.Queues;
-import reactor.util.retry.Retry;
 
 import java.io.Closeable;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Queue;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
+import static com.azure.core.amqp.implementation.RetryUtil.withRetry;
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.EMIT_RESULT_KEY;
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.PARTITION_ID_KEY;
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.SIGNAL_TYPE_KEY;
+
 /**
  * Keeps track of publishing events to a partition.
  */
 class EventHubBufferedPartitionProducer implements Closeable {
-    private final ClientLogger logger;
+    private static final ClientLogger LOGGER = new ClientLogger(EventHubBufferedPartitionProducer.class);
+
+    private final AmqpRetryOptions retryOptions;
     private final EventHubProducerAsyncClient client;
     private final String partitionId;
     private final AmqpErrorContext errorContext;
     private final AtomicBoolean isClosed = new AtomicBoolean(false);
     private final Disposable publishSubscription;
     private final Sinks.Many<EventData> eventSink;
-    private final Retry retryWhenPolicy = Retry.from(signal -> {
-        if (isClosed.get()) {
-            return Mono.empty();
-        } else {
-            return Mono.just(true);
-        }
-    });
     private final CreateBatchOptions createBatchOptions;
     private final Queue<EventData> eventQueue;
+    private final AtomicBoolean flush = new AtomicBoolean(false);
```

Comment:
we have a function `flush()`, change to `isFlushed` for boolean name?

Reply:
When I was naming this variable, I kept thinking of the bathroom. 😄 But you're right, to align, I'll rename this.